src/LightSaml/SpBundle/Security/Firewall/LightSamlLogoutHandler.php

```diff
@@ -0,0 +1,196 @@
+ * with this source code in the file LICENSE.
+ */
+
+namespace LightSaml\SpBundle\Security\Firewall;
+
+use LightSaml\Binding\BindingFactoryInterface;
+use LightSaml\Context\Profile\MessageContext;
+use LightSaml\Model\Protocol\LogoutRequest;
+use LightSaml\Model\Protocol\LogoutResponse;
+use LightSaml\SamlConstants;
+use LightSaml\SpBundle\Model\Protocol\LogoutMessageContextFactory;
+use LightSaml\State\Sso\SsoSessionState;
+use LightSaml\Store\Sso\SsoStateStoreInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Http\Logout\LogoutSuccessHandlerInterface;
+
+/**
+ * Class LightSamlLogoutHandler.
+ */
+class LightSamlLogoutHandler implements LogoutSuccessHandlerInterface
+{
+    /** @var BindingFactoryInterface */
+    private $bindingFactory;
+    /** @var SessionInterface */
+    private $session;
+    /** @var SsoStateStoreInterface */
+    private $ssoStateStore;
+    /** @var RouterInterface */
+    private $router;
+    /** @var LogoutMessageContextFactory */
+    private $logoutMessageContextFactory;
+
+    /**
+     * LightSamlLogoutHandler constructor.
+     *
+     * @param BindingFactoryInterface     $bindingFactory
+     * @param SsoStateStoreInterface      $ssoStateStore
+     * @param RouterInterface             $router
+     * @param LogoutMessageContextFactory $logoutMessageContextFactory
+     */
+    public function __construct(
+        BindingFactoryInterface $bindingFactory,
+        SsoStateStoreInterface $ssoStateStore,
+        RouterInterface $router,
+        LogoutMessageContextFactory $logoutMessageContextFactory
+    ) {
+        $this->bindingFactory = $bindingFactory;
+        $this->ssoStateStore = $ssoStateStore;
+        $this->router = $router;
+        $this->logoutMessageContextFactory = $logoutMessageContextFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function onLogoutSuccess(Request $request)
+    {
+        $this->session = $request->getSession();
+        $samlMessage = $this->getIncomingMessage($request);
+
+        if (null === $samlMessage) {
+            return $this->sendLogoutRequestToIP();
+        }
+
+        if ($samlMessage instanceof LogoutResponse) {
+            return $this->handleLogoutResponseFromIP($samlMessage);
+        } elseif ($samlMessage instanceof LogoutRequest) {
+            return $this->handleLogoutRequestFromIP($samlMessage);
+        }
+
+        throw new InvalidSamlMessageForLogoutException($samlMessage);
+    }
+
+    /**
+     * @return Response
+     */
+    private function sendLogoutRequestToIP()
+    {
+        $sessions = $this->ssoStateStore->get()->getSsoSessions();
+
+        if (count($sessions) === 0) {
+            return $this->createLogoutResponse();
```

Comment:
This method is called when there's no incoming message - thus IDP didn't sent a ``LogoutRequest``. But, here, in case there are no sso sessions in store, we're making ``LogoutResponse`` that will get rejected by IDP. Isn't an exception better suited?